File: app/src/main/java/com/uvnode/typhoon/extensions/source/Filterable.java
package com.uvnode.typhoon.extensions.source;

import com.uvnode.typhoon.extensions.api.ApiCallbacks;
import com.uvnode.typhoon.extensions.model.Filter;

import java.util.ArrayList;

public interface Filterable {

    ArrayList<Filter> getFilters();

    void search(ApiCallbacks callbacks, ArrayList<Filter> filters, int page) throws Exception;
}

File: app/src/main/java/com/uvnode/typhoon/extensions/source/Filterable.kt
package com.uvnode.typhoon.extensions.source

import com.uvnode.typhoon.extensions.api.ApiCallbacks
import com.uvnode.typhoon.extensions.model.Filter

interface Filterable {
    val filters: List<Filter<*>?>?

    @Throws(Exception::class)
    fun search(callbacks: ApiCallbacks, filters: List<Filter<*>?>, page: Int)
}